set.mm

```diff
@@ -215251,6 +215253,65 @@ is an important property of monoids (see ~ mndid ), and therefore also for
       IUGUJUKULUMUNAHUOTUPUSVEACVGUQTUR $.
   $}
 
+  ${
+    $d n u v w x y z B $.  $d n u v w x y z O $.  $d n u v w x y z ph $.
+    $d u v w y z N $.  $d n u v w x y z .+ $.  $d u v w y z X $.
+    $d u v w y ps $.
+    grprinvlem.c $e |- ( ( ph /\ x e. B /\ y e. B ) -> ( x .+ y ) e. B ) $.
+    grprinvlem.o $e |- ( ph -> O e. B ) $.
+    grprinvlem.i $e |- ( ( ph /\ x e. B ) -> ( O .+ x ) = x ) $.
+    grprinvlem.a $e |- ( ( ph /\ ( x e. B /\ y e. B /\ z e. B ) )
+          -> ( ( x .+ y ) .+ z ) = ( x .+ ( y .+ z ) ) ) $.
+    grprinvlem.n $e |- ( ( ph /\ x e. B ) -> E. y e. B ( y .+ x ) = O ) $.
+    ${
+      grprinvlem.x $e |- ( ( ph /\ ps ) -> X e. B ) $.
+      grprinvlem.e $e |- ( ( ph /\ ps ) -> ( X .+ X ) = X ) $.
+      $( Lemma for ~ grprinvd .  (Contributed by NM, 9-Aug-2013.) $)
+      grprinvlem $p |- ( ( ph /\ ps ) -> X = O ) $=
+        ( cv co wceq wcel vu vv vw wa wrex wral ralrimiva oveq2 rexbidv cbvralv
+        eqeq1d sylib syl2an2r oveq2d adantr simprr oveq1d w3a caovassg ad4ant14
+        rspccva simprl caovassd id eqeq12d rspcdva 3eqtr3d rexlimddv ) ABUDZDQZ
+        IGRZHSZIHSDFAVJEQZGRZHSZDFUEZEFUFZBIFTZVLDFUEZAVJCQZGRZHSZDFUEZCFUFVQAW
+        CCFNUGWCVPCEFVTVMSZWBVODFWDWAVNHVTVMVJGUHUKUIUJULOVPVSEIFVMISZVOVLDFWEV
+        NVKHVMIVJGUHUKUIVAUMVIVJFTZVLUDZUDZVJIIGRZGRZVKIHVIWJVKSWGVIWIIVJGPUNUO
+        WHVKIGRHIGRZWJIWHVKHIGVIWFVLUPZUQWHUAUBUCVJIIFGAUAQZFTUBQZFTUCQZFTURWMW
+        NGRWOGRWMWNWOGRGRSBWGACDEWMWNWOFGMUSUTVIWFVLVBVIVRWGOUOZWPVCVIWKISZWGVI
+        HVJGRZVJSZWQDFIVJISZWRWKVJIVJIHGUHWTVDVEAWSDFUFZBAHVTGRZVTSZCFUFXAAXCCF
+        LUGXCWSCDFVTVJSZXBWRVTVJVTVJHGUHXDVDVEUJULUOOVFUOVGWLVGVH $.
+    $}
+
+    ${
+      grprinvd.x $e |- ( ( ph /\ ps ) -> X e. B ) $.
+      grprinvd.n $e |- ( ( ph /\ ps ) -> N e. B ) $.
+      grprinvd.e $e |- ( ( ph /\ ps ) -> ( N .+ X ) = O ) $.
+      $( Deduce right inverse from left inverse and left identity in an
+         associative structure (such as a group).  (Contributed by NM,
+         10-Aug-2013.)  (Proof shortened by Mario Carneiro, 6-Jan-2015.) $)
```

Comment:
Proposal:

> In a semigroup with a left identity and right inverses, any right inverse is also a left inverse to the same element.

Reply:
Semigroups are not defined yret (later in section "Semigroups", and I do not want (and see no need) to change Norm's comment here.